teps/0094-configuring-resources-at-runtime.md

```diff
@@ -52,12 +61,495 @@ While this may be a valuable feature, it should be considered in a separate prop
+
+Some users may want to have resource requirements apply to every `Step` or `Sidecar`.
+This TEP proposes using "\*" as a special string which applies the 
+specified resource requirements to all `Step`s or `Sidecar`s. "\*" is not a valid `Container` name.
+If "\*" is used, it must be the only key present in `StepResources` or `SidecarResources`.
+This will be the only way to override resource requirements of unnamed `Step`s or `Sidecar`s.
+
+An alternative option is to use a repeated field for container overrides, where the index
+of the container override is the same as the index of the `Step` or `Sidecar` it applies to.
+This strategy could result in unintended overrides in a `TaskRun`
+if `Step`s are reordered, added, or removed in the corresponding `Task`.
+
+#### Merging resource requirements
+`Step` resource requirements can currently be specified in both
+`Task.Step.Resources` and `Task.StepTemplate.Resources`.
+If resource requirements are specified in both fields, the value present in `Task.Step.Resources`
+is used. However, different resource types (e.g. CPU, memory) are considered independently.
+For example:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: image-build-task
+spec:
+  steps:
+    - name: build
+      resources:
+        requests:
+          memory: 500Mi
+        limits:
+          memory: 600Mi
+  stepTemplate:
+    resources:
+      requests:
+        memory: 300Mi
+        cpu: 0.5
+```
+A resulting `TaskRun` will have a memory request of 500Mi, a memory limit of 600Mi,
+and a CPU request of 0.5 CPU units.
+
+This proposal adds a third way to specify `Step` resource requirements: `TaskRun.StepResources`.
+`TaskRun.StepResources` will override `Task.Step.Resources` in the same way that that field
+overrides `Task.StepTemplate.Resources`.
+Using the example `Task` defined above, consider the following `TaskRun`:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: image-build-taskrun
+spec:
+  taskRef:
+    name: image-build-task
+  stepResources:
+    build:
+      resources:
+        requests:
+          memory: 1Gi
+```
+The `TaskRun` will have a memory request of 1Gi and no memory limit, as this is the configuration
+specified for memory in `TaskRun.StepResources`. It will have a CPU request of 0.5 CPU units and no
+CPU limit, as this configuration was specified in the `Task.StepTemplate.Resources`
+and not overridden by the `Task.Step.Resources` or `TaskRun.StepResources`.
+
+#### Alternative Syntax
+
+Instead of using the Kubernetes `ResourceRequirements` directly, we could create a
+Tekton-owned struct for container overrides. This could be useful if we anticipate a future
+need to override additional `Container` fields in TaskRuns.
+
+However, at the moment, there is no clear use case for such features,
+and the ability to override `Step` and `Sidecar` fields
+other than `Resources` is a non-goal of this proposal.
```

Comment:
I think the use of named sub-objects addresses this concern, because we're now using a Tekton-owned struct that we can easily add fields to. Feel free to reopen this comment if your concern hasn't been addressed!